blog/2015-03-30-Eclipse_Docker_Tooling.adoc

```diff
@@ -0,0 +1,100 @@
+= Upcoming Docker Tooling for Eclipse
+:page-layout: blog
+:page-author: xcoulon
+:page-tags: [jbosstools, docker, eclipse]
+
+JBoss Tools and Eclipse Linux Tools team members from Red Hat are currently busy working on
+Docker tooling for Eclipse and even though the code is not public yet (more details on that at
+the end of this blog entry), I thought it would be interesting to share our progress on the project.
+
+The Docker tooling is aimed at providing at minimum the same basic level features as the command-line
+interface, but also provide some advantages by having access to a full fledged UI.
+
+== Docker Explorer
+
+The *Docker Explorer* provides a wizard to establish a new connection to a Docker daemon.
+This wizard can detect default settings if the user's machine runs Docker natively or in a VM using Boot2Docker.
+Both Unix sockets on Linux machines and the REST API on other OSes are detected and supported.
+The wizard also allows remote connections using custom settings.
+
+image::images/docker_tooling_connection_wizard.png[Connection wizard]
+
+The *Docker Explorer* itself is a tree view that handles multiple connections and provides users
+with quick overview of the existing images and containers.
+
+image::images/docker_tooling_explorer_view.png[Docker Explorer view]
+
+Built-in filters can show/hide intermediate and 'dangling' images as well as
+stopped containers.
+
+== Managing Docker Images
+
+The *Docker Images view* lists all images in the Docker host selected in the
+Docker Explorer view. This view allows user to manage images, including:
+
+- Pulling images from the https://registry.hub.docker.com/[Docker Hub Registry] (other registries will be supported as well)
+- Uploading images to the https://registry.hub.docker.com/[Docker Hub Registry]
+- Building images from a Docker files
+- Running an image
+
+image::images/docker_tooling_images_view.png[Docker Images view]
+
+A wizard let the user input all the arguments to create a new container from an image.
+When the container is started, all the logs can be streamed into the Eclipse Console:
+
+image::images/docker_tooling_console.png[Docker Images view]
+
+== Managing Docker Containers
+
+The *Docker Containers* view lets the user manage her containers. The view toolbar
+provides commands to start, stop, pause, unpause, attach and kill containers.
+
+image::images/docker_tooling_containers_view.png[Docker Containers view]
+
+This view also provides a filter to show/hide stopped containers.
+Users can also attach an Eclipse console to a running Docker container to follow
+the logs and use the STDIN to interact with it.
```

Comment:
We only allow STDIN if the container is set up with openStdin ahead of time.  I am not sure if we can attach to an existing container and set up STDIN support.